### PR TITLE
Create signature for Updater

### PR DESCRIPTION
**Pull request by apelisse**
Create the signature for Updater, the object that will be used to update
the owners Fieldset and merge applied configurations.

### CODE REVIEW

**Comment by apelisse** [comment]
@lavalamp @kwiesmueller

File: merge/update.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+// Updater is the object used to compute updated FieldSets and also
+// merge the object on Apply.
+type Updater struct{}
+
+// Update is the method you should call once you've merged your final
+// object on CREATE/UPDATE/PATCH verbs. newObject must be the object
+// that you intend to persist, and liveObject must be the original
+// object. owners is modified and needs to be injected in the object
+// before persisting.
+func (s *Updater) Put(liveObject, newObject typed.TypedValue, owners Owners, owner string) error {
```
Comment (kwiesmueller) [inline merge/update.go]: Should be Update or the comment should be different

File: merge/update.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+type Owners map[string]*VersionedSet
+
+// Updater is the object used to compute updated FieldSets and also
+// merge the object on Apply.
+type Updater struct{}
+
+// Update is the method you should call once you've merged your final
+// object on CREATE/UPDATE/PATCH verbs. newObject must be the object
+// that you intend to persist, and liveObject must be the original
+// object. owners is modified and needs to be injected in the object
```
Comment (lavalamp) [inline merge/update.go]: Is it better to modify owners or construct & return a new one? I think most of the types in this repo behave better if treated as immutable. (same question on `Apply`, too)

File: merge/update.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+// what version).
+type Owners map[string]*VersionedSet
+
+// Updater is the object used to compute updated FieldSets and also
+// merge the object on Apply.
+type Updater struct{}
+
+// Update is the method you should call once you've merged your final
+// object on CREATE/UPDATE/PATCH verbs. newObject must be the object
+// that you intend to persist, and liveObject must be the original
```
Comment (lavalamp) [inline merge/update.go]: that you intend to persist `(after applying the patch if this is for a PATCH call)`, and liveObject must be the original object `(empty if this is a CREATE call)`.

File: merge/update.go (hunk 2 above)
Comment (apelisse) [reply merge/update.go]: Yeah, so we don't have a good mechanism to change the typedvalue right now, but I think it makes more sense to return the object with the owners field set already.

**Comment by apelisse** [comment]
PTAL

**Comment by kwiesmueller** [comment]
After looking at it again I wanted to ask why Owners isn't part of the Object being updated (which would make returning them easier as well).

**Comment by apelisse** [comment]
> After looking at it again I wanted to ask why Owners isn't part of the Object being updated (which would make returning them easier as well).

I remember why now. The reason is that the types are not exactly the same. What we will store in the API is not the same type that we use to do the set operations. Which means we need a layer to convert from one to another, and I don't think that layer will live in this repository.

**Pull request merged**